gprofiler/profilers/perf.py

```diff
@@ -158,6 +165,7 @@ def __init__(
     ):
         super().__init__(frequency, duration, stop_event, storage_dir)
         self._perfs: List[PerfProcess] = []
+        self._golang_metadata = GolangMetadata(stop_event)
```

Comment:
Move Golang related code to `SystemProfilerWithGolang(SystemProfiler)`, that's a good case for extending the class. (name is up to you :)

Reply:
(fix the usage in gProfiler to use the new `SystemProfilerWithGolang`)

Reply:
Actually, it's not very convenient with the profilers registry, so meh. If u find another way to encapsulate the golang code in another class, good. Otherwise meh